docs/source/developer/accessibility.rst

```diff
@@ -0,0 +1,278 @@
+
+Therefore there is an effort underway to use `Playwright
+<https://playwright.dev>`__ to write user-level `accessibility tests to
+JupyterLab
+<https://github.com/Quansight-Labs/jupyter-a11y-testing/tree/main/testing/jupyterlab>`__.
+To illustrate how to use it within your development process, let's walk through
+an example.
+
+This example will involve three separate GitHub repos:
+
+1. `Quansight-Labs/jupyter-a11y-testing
+   <https://github.com/Quansight-Labs/jupyter-a11y-testing>`__
+2. `jupyterlab/lumino <https://github.com/jupyterlab/lumino>`__
+3. `jupyterlab/jupyterlab <https://github.com/jupyterlab/jupyterlab>`__
+
+This is a real world example, taken from actual past work.
+
+Let's say you do an accessibility audit of the start page of the JupyterLab UI
+and find a tab trap in the top menu bar, meaning the user can press the tab key
+to get into the menu bar but cannot easily get past it using only the keyboard.
+
+You dig in further and discover that the `tab trap bug is in the
+jupyterlab/lumino repo <https://github.com/jupyterlab/lumino/pull/373>`__, so
+you fork the jupyterlab/lumino repo, create a new branch called
+``fix-tab-trap``, and open a pull request.
+
+You decide that you want to write a test. This is one of those cases where you
+can fairly easily write a unit test. However, a unit test would only check the
+top menu bar, so it would not prevent a reappearance of the issue that you
+decided you want to fix once and for all, namely: you don't want any tab traps
+anywhere on the JupyterLab start page.
+
+So you decide that you want to `add a regression test to the
+Quansight-Labs/jupyter-a11y-testing repo
+<https://github.com/Quansight-Labs/jupyter-a11y-testing/blob/f36bf5b2e8cb87613c637fc5aa03401c92ec58d0/testing/jupyterlab/tests/regression-tests/no-tab-trap-initial-page.test.ts>`__.
+This test checks that there are no tab traps on the JupyterLab start page by
+using Playwright to open JupyterLab and press the tab key repeatedly. So as with
+the Lumino repo before, you fork the Quansight-Labs/jupyter-a11y-testing repo,
+create a branch called ``test-tab-trap``, and open a pull request. The important
+thing in this step is that you save your test file with a ``.test.ts`` extension
+next to the other regression test files.
+
+Now you want to run your test. Specifically, you want to run the test against a
+build of JupyterLab that incorporates your Lumino fix. Here's how you would do
+that.
+
+Let's pretend that your GitHub username is *a11ydev* and you've forked the
+Lumino and testing repos and created the following branches on those forks, one
+with your bug fix and the other with your test:
+
+1. ``a11ydev/lumino:fix-tab-trap``
+2. ``a11ydev/jupyter-a11y-testing:test-tab-trap``
+
+On GitHub, go to your fork of the testing repo, *a11ydev/jupyter-a11y-testing*.
+Make sure that you are on your `test-tab-trap` branch, which contains the
+.test.ts file that you added. Then go to Actions and click on the workflow
```

Comment:
```suggestion
``.test.ts`` file that you added. Then go to Actions and click on the workflow
```

Reply:
I don't really want to call attention to this string, though. It's not actually a literal. If it were the actual literal file name, then I would want to format it this way. Should I put it in quotes or italics instead?

Reply:
Oh nevermind, I see that this is for consistency